base/system/winlogon/sas.c

```diff
@@ -795,6 +795,11 @@ HandleLogoff(
     DWORD exitCode;
     NTSTATUS Status;
 
+    /* HACK & FIXME: Prevents logging off a user with a NULL token (e.g. SYSTEM pseudo account).
+     * However this appears to happen too early. */
+    if (Session->UserToken == NULL)
+        return STATUS_INVALID_PARAMETER;
```

Comment:
Does it happen too early as in what exactly?

Since it's yet to investigate if the session's token could be set to NULL for other reasons other than being in SYSTEM's security context, checking if the ROS build is a LiveCD one as I mentioned above is more **_correct_** in the sense that you're 100% sure that indeed the ROS build is a LiveCD one and the running security context of the current session is SYSTEM, instead of preventing a log off by checking the status of the session alone.

Reply:
It doesn't sound like checking for LiveCD --> you're SYSTEM is the best way. One can make a ROS (or Windows) installation to run as system (e.g. the sethc.exe hack, or having a cmd.exe launcher as a LocalService service, that you start from the services manager, and from where you can execute explorer.exe, etc.)